avocado/plugins/yaml_to_mux.py

```diff
@@ -432,7 +538,11 @@ def run(self, args):
                 entry.insert(0, '')  # add path='' (root)
             data.get_node(entry[0], True).value[entry[1]] = entry[2]
 
+        mux_path = getattr(args, 'mux_path', None)
+        if mux_path is None:
+            mux_path = ['/run/*']
+
         data = apply_filters(data, getattr(args, 'mux_filter_only', None),
                              getattr(args, 'mux_filter_out', None))
         if data != MuxTreeNode():
-            args.mux.data_merge(data)
+            args.mux.add_variants_plugin(MuxPlugin(data, mux_path))
```

Comment:
Yes, this is just a prep-work. I think it's more important to get this working as it improves the handling and then continue improving it. First we need to know, whether it's possible to separate it and when we have somehow stable API we should focus on making it properly modular. Defining single `add_*_plugin` should be enough to then turn it into a proper plugin and I think it's fine for now.